pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,184 @@
+      {
+        "requires": ["pywin32 > 1.0"],
+        "environment": "sys.platform == 'win32'"
+      }
+    ]
+
+
+Micro-language
+==============
+
+The micro-language behind this is a simple subset of Python. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolen operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is not ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (SUBEXPR [CMPOP SUBEXPR])
+    CMPOP: (==|!=|<|>|<=|>=|in|not in)
+
+but my memory of BNF isn't quite up to fixing it just now.
+
+``SUBEXPR`` is either a Python string (such as ``'2.4'``, or
+``'win32'``) or one of the below marker variables.
+
+Version numbers
+---------------
+
+* ``python_version``: ``'{0.major}.{0.minor}'.format(sys.version_info)``
+* ``python_full_version``: see definition below
+* ``platform_version``: ``platform.version()``
+* ``implementation_version````: see definition below
+
+All version numbers must be valid PEP-0440[3] identifiers, and will be
+compared using PEP-0440 semantics.
+
+Strings
+-------
+
+* ``os_name````: ``os.name``
+* ``sys_platform````: ``sys.platform``
+* ``platform_release``: ``platform.release()``
+* ``implementation_name````: ``sys.implementation.name``
+* ``platform_machine``: ``platform.machine()``
+* ``platform_python_implementation``: ``platform.python_implementation()``
+
+
+If a particular value is not available (such as the ``sys.implementation``
+subattributes in versions of Python prior to 3.3), the corresponding marker
+variable MUST be considered equivalent to the empty string.
+
+Note that it is not possible to use other sequences like tuples or
+lists on the right side of the ``in`` and ``not in`` operators,
+regardless of whether the comparison is with a version number, string,
+or one of the marker variable names (which either refers to a string
+or a variable number)
+
+Chaining of comparison operations is permitted using the normal Python
+semantics of an implied ``and``.
+
+
+Version information variables
+=============================
+
+The ``python_full_version`` and ``implementation_version`` marker variables
+are derived from ``sys.version_info()`` and ``sys.implementation.version``
+respectively, in accordance with the following algorithm::
+
+    def format_full_version(info):
+        version = '{0.major}.{0.minor}.{0.micro}'.format(info)
+        kind = info.releaselevel
+        if kind != 'final':
+            version += kind[0] + str(info.serial)
+        return version
+
+    python_full_version = format_full_version(sys.version_info)
+    implementation_version = format_full_version(sys.implementation.version)
+
+``python_full_version`` will typically correspond to the leading segment
+of ``sys.version()``.
```

Comment:
I've tweaked this to use sys.version and sys.version_info, and to give an actual working python definition of python_full_version.

I'm not sure that these changes have actually clarified things.